src/common/src/row/compacted_row.rs

```diff
@@ -20,20 +22,20 @@ use crate::util::value_encoding;
 /// Executors need to serialize Row into `CompactedRow` before writing into cache.
 #[derive(Clone, Debug, Default, PartialEq, Eq, Hash)]
 pub struct CompactedRow {
-    pub row: Vec<u8>,
+    pub row: Bytes,
 }
 
 impl CompactedRow {
     /// Deserialize [`CompactedRow`] into [`OwnedRow`] with given types.
     pub fn deserialize(&self, data_types: &[DataType]) -> value_encoding::Result<OwnedRow> {
-        RowDeserializer::new(data_types).deserialize(self.row.as_slice())
+        RowDeserializer::new(data_types).deserialize(self.row.as_ref())
     }
 }
 
 impl<R: Row> From<R> for CompactedRow {
     fn from(row: R) -> Self {
         Self {
-            row: row.value_serialize(),
+            row: Bytes::from(row.value_serialize()),
```

Comment:
Will this lead to allocation? IIUC, the allocation is caused by `Vec::shrink_to_fit`. 🤔

Reply:
I have change some of our serialization implementation to return `Bytes` directly to avoid first creating a `Vec` and then do the conversion which may cause memory allocation. The code is using `BytesMut` to build the `Bytes`.

Reply:
I'm not quite sure but the underlying container of `BytesMut` is `Vec` as well? 🤔

Reply:
Seems not. The `BytesMut` implements its own container with pointer and len. And the `freeze` method is said to be of zero-cost in the doc. https://docs.rs/bytes/latest/bytes/struct.BytesMut.html#method.freeze